test/Npgsql.Tests/ConnectionTests.cs

```diff
@@ -1308,6 +1308,23 @@ public async Task TcpKeepalive()
                 Thread.Sleep(Timeout.Infinite);
         }
 
+        [Test, Description("Asserting TCP keepalive time and interval setting does not throw exceptions")]
+        public async Task TcpKeepaliveSupport()
+        {
+            var csb = new NpgsqlConnectionStringBuilder(ConnectionString)
+            {
+                TcpKeepAliveTime = 2000,
+                TcpKeepAliveInterval = 3000,
+            };
+
+            using (var conn = OpenConnection(csb))
+            using (var cmd = new NpgsqlCommand("SELECT 1", conn))
```

Comment:
Since this test doesn't actually check that keepalives works - only that setting the options doesn't cause an exception - we can remove the actual command execution. Since the actual value of the test is quite limited, we can also remove it entirely.

If you do want to keep the test, note that the physical connection will remain open forever, since pooling is on. See other tests for how CreateTempPool is used to clear the pool at the end of the test.

Reply:
> I don't have a particular objection to merging this, but I also don't think it's going to help a lot of people :) Let me know what you think.

We are "pre- .NET Core 3.1 users" and this will help anyone using Mono :)


> Since this test doesn't actually check that keepalives works - only that setting the options doesn't cause an exception - we can remove the actual command execution. Since the actual value of the test is quite limited, we can also remove it entirely.
> 
> If you do want to keep the test, note that the physical connection will remain open forever, since pooling is on. See other tests for how CreateTempPool is used to clear the pool at the end of the test.

I'm understood, thank you!
Test removed.